### PR TITLE
webui: do not apply  log_step decorator for non permian tests

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

**Comment by velezd** [comment]
I would suggest renaming the tests and class to EndToEnd tests or something like that. Because they can be executed without Permian and Permian has workflows for other kinds of tests not just for the WebUI.

File: ui/webui/test/helpers/step_logger.py [hunk 1]
```diff
@@ -23,6 +24,8 @@ def log_step(snapshots=False, snapshot_before=False, snapshot_after=False, docst
     :type docstring: bool, optional
     """
     def decorator(function):
+        if not os.environ.get('PERMIAN', False):
```
Comment (velezd) [inline ui/webui/test/helpers/step_logger.py:27]: The ENV variable name should be different, it has nothing to do with Permian. And I think it should only have an effect on the creation of the screenshots. Logging what steps are being performed is always useful and AFAIK has no negative effects.

**Review by velezd — CHANGES_REQUESTED** [review]

**Comment by KKoukiou** [comment]
So removing the step logger decorator changed the timing and revealed some flakiness, which makes 2 tests now consistently fail on this PR. Will debug this here.

File: ui/webui/test/helpers/step_logger.py (hunk 1 above)
Comment (M4rtinK) [reply ui/webui/test/helpers/step_logger.py]: Yes & more detailed logs are especially useful when we hit race conditions.

File: ui/webui/test/helpers/step_logger.py [hunk 2]
```diff
@@ -34,13 +34,16 @@ def wrapper(*args, **kwargs):
             print(f'[TEST STEP] {function.__name__}{nice_args}')
             if docstring:
                 print(f'[DOC] {function.__doc__}')
-            if snapshots or snapshot_before:
-                BrowserSnapshot.new(args[0].browser)
+
+            if os.environ.get('END2END', False):
```
Comment (velezd) [inline ui/webui/test/helpers/step_logger.py:38]: I suggest doing it like this:
```
end2end = os.environ.get('END2END', False)
if end2end and (snapshots or snapshot_before):
    ...
if end2end and (snapshots or snapshot_after):
    ...
```

**Review by velezd — CHANGES_REQUESTED** [review]

File: ui/webui/test/helpers/step_logger.py (hunk 2 above)
Comment (KKoukiou) [reply ui/webui/test/helpers/step_logger.py]: Adjusted.

**Review by velezd — APPROVED** [review]

**Comment by KKoukiou** [comment]
testing farm is broken because of dnf5 https://redhat-internal.slack.com/archives/C04MU29TES1/p1687419798774249

**Pull request merged**